### PR TITLE
clusters: add Control Plane nodes widget to the MAPI cluster detail page

### PR DESCRIPTION
**Pull request by axbarsan**
Towards giantswarm/giantswarm#9077

This PR adds a new widget for displaying info about a cluster's Control Plane nodes, via the Management API. In the spec it was originally called `Master nodes`, but from my understanding, we changed that wording to `Control Plane nodes` everywhere.

## Preview

<details>
<summary>1/1 nodes ready</summary>

image

</details>

<details>
<summary>0/3 nodes ready</summary>

image

</details>

<details>
<summary>1/3 nodes ready</summary>

image

</details>

<details>
<summary>3/3 nodes ready</summary>

image

</details>

<details>
<summary>loading state</summary>

image

</details>

<details>
<summary>nodes not available (e.g. an error occured)</summary>

image

</details>

### CODE REVIEW

File: src/components/MAPI/types.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import * as capzv1alpha3 from 'model/services/mapi/capzv1alpha3';
+
+export type ControlPlaneNode = capzv1alpha3.IAzureMachine;
+
+export type ControlPlaneNodeList = capzv1alpha3.IAzureMachineList;
```
Comment (axbarsan) [inline src/components/MAPI/types.ts:5]: Created a type union for all control plane node types

File: src/components/MAPI/utils.ts [hunk 2]
```diff
@@ -181,6 +183,7 @@ export async function fetchMasterListForCluster(
         labelSelector: {
           matchingLabels: {
             [capiv1alpha3.labelCluster]: cluster.metadata.name,
+            [capzv1alpha3.labelControlPlane]: 'true',
```
Comment (axbarsan) [inline src/components/MAPI/utils.ts:186]: In the current REST API logic, this is also necessary in order for an `AzureMachine` to be a Control Plane node.

File: src/model/services/mapi/capiv1alpha3/key.ts [hunk 3]
```diff
@@ -41,26 +42,32 @@ export function getKubernetesAPIEndpointURL(
   return `https://${hostname}`;
 }
 
+interface IConditionGetter {
+  status?: {
+    conditions?: ICondition[];
+  };
+}
+
 export function getCondition(
```
Comment (axbarsan) [inline src/model/services/mapi/capiv1alpha3/key.ts:51]: Thanks to duck typing we can make these condition helpers more generic, and use them for multiple resources

**Comment by marians** [comment]
> In the spec it was originally called `Master nodes`, but from my understanding, we changed that wording to `Control Plane nodes` everywhere.

That's right, there is https://github.com/giantswarm/giantswarm/issues/14964

I'm fine with starting here. We still might have to do a bit more in happa.

I noticed that I would like to change the spec a bit. How about using a `<code>Ready</code>` literal for the state name, to indicate that it's about the Kubernetes node state `Ready` and not any other readiness definition.

**Review by marians — APPROVED** [review]

**Pull request merged**